docs/modules/ROOT/pages/access.adoc

```diff
@@ -1,672 +1,415 @@
+mod Contract {
+    use starknet::ContractAddress;
+    use super::Ownable;
 
-To restrict a function's access to the owner only, add in the `assert_only_owner` method:
+    #[storage]
+    struct Storage {}
 
-[,cairo]
-----
-from openzeppelin.access.ownable.library import Ownable
+    #[constructor]
+    fn constructor(self: @ContractState, owner: ContractAddress) {
+        let mut unsafe_state = Ownable::unsafe_new_contract_state();
+        Ownable::InternalImpl::initializer(ref unsafe_state, owner);
+    }
 
-func protected_function{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
-    Ownable.assert_only_owner();
-    return ();
+    (...)
 }
 ----
 
-=== Ownable library API
+To restrict a function's access to the owner only, add in the `assert_only_owner` method:
 
-[,cairo]
+[,javascript]
 ----
-func initializer(owner: felt) {
-}
-
-func assert_only_owner() {
-}
-
-func owner() -> (owner: felt) {
-}
+#[starknet::contract]
+mod Contract {
+    use openzeppelin::access::ownable::Ownable;
 
-func transfer_ownership(new_owner: felt) {
-}
+    (...)
 
-func renounce_ownership() {
-}
+    #[external(v0)]
+    fn foo(ref self: ContractState) {
+        // This function can only be called by the owner
+        let unsafe_state = Ownable::unsafe_new_contract_state();
+        Ownable::InternalImpl::assert_only_owner(@unsafe_state);
 
-func _transfer_ownership(new_owner: felt) {
+        (...)
+    }
 }
 ----
 
-==== `initializer`
-
-Initializes Ownable access control and should be called in the implementing contract's constructor.
-Assigns `owner` as the initial owner address of the contract.
-
-This must be called only once.
-
-Parameters:
-
-[,cairo]
-----
-owner: felt
-----
-
-Returns: None.
-
-==== `assert_only_owner`
-
-Reverts if called by any account other than the owner.
-In case of renounced ownership, any call from the default zero address will also be reverted.
-
-Parameters: None.
-
-Returns: None.
-
-==== `owner`
-
-Returns the address of the current owner.
-
-Parameters: None.
-
-Returns:
-
-[,cairo]
-----
-owner: felt
-----
-
-==== `transfer_ownership`
-
-Transfers ownership of the contract to a new account (`new_owner`).
-Can only be called by the current owner.
-
-Emits a <<ownershiptransferred,`OwnershipTransferred`>> event.
-
-Parameters:
-
-[,cairo]
-----
-new_owner: felt
-----
-
-Returns: None.
-
-==== `renounce_ownership`
-
-Leaves the contract without owner.
-It will not be possible to call functions with `assert_only_owner` anymore.
-Can only be called by the current owner.
-
-Emits a <<ownershiptransferred,`OwnershipTransferred`>> event.
-
-Parameters: None.
-
-Returns: None.
-
-[#transfer-ownership-internal]
-==== `_transfer_ownership`
+=== Interface
 
-Transfers ownership of the contract to a new account (`new_owner`). {extensibility-pattern}[`internal`] function without access restriction.
+This is the full interface a contract must implement to be compliant with the `Ownable` implementation:
```

Comment:
```suggestion
This is the full interface of the `Ownable` implementation:
```
implementations are not specs, i insist it doesn't make sense to say something is "compliant with the implementation"